### ISSUE TITLE
It looks like Ocaml's new "raw string" syntax isn't supported by camlp5

### ISSUE THREAD

**Issue by chetmurthy** [camlp5/camlp5#33]
Daniel,

I'm using ocaml 4.07.1, camlp5 7.06.10-g84ce6cc4, and have looked in the camlp5 source from github (lib/plexer.ml ??) and it appears that ocaml's new syntaxes for raw strings, viz

{|argle bargle|}

and

{foo|argle bargle|foo}

are not supported in camlp5.

I think the problem in that file is that (1) "{|" and "|}" are currently keywords, and (2) the current string lexing code doesn't parse the new extended syntax.

I believe I can fix these issues, but am unsure of what path to take, or if you'd rather address them yourself.  If you can let me know of your preferred solution, I'll attempt to implement the fix and send you a PR.

My own inclination would be that I leave "{|" and "|}" as keywords, and just add support for the second form of raw strings -- if somebody's using a raw string, it's not much to ask them to use "{boo|...|boo}", rather than "{|....|}".  But perhaps you'd prefer to remove those two keywords?

**Comment by chetmurthy** [comment]
I think I've got a working fix for this issue.  I've written the code so that (as described above) raw strings MUST be of the form "{boo|....|boo}" (cannot be "{|...|}") so that "{|" and "|}" remain tokens.  Also, when pretty-printing back out, the literal is converted into normal Ocaml string syntax.

The only snag is that because camlp5's lexer is an LL(1) lexer, I have to use "npeek" lookahead to recognize the start of a raw-string-literal.

Also, I'm new to hacking camlp5's codebase, so I don't know if I did things correctly.  I can send a PR?
